binderhub/static/js/index.js

```diff
@@ -90,36 +90,36 @@ Image.prototype.close = function() {
         this.eventSource.close();
     }
 };
-
-Image.prototype.launch = function(url, token, filepath, pathType) {
+Image.prototype.launch = function(url, token, path, pathType) {
     // redirect a user to a running server with a token
-    if (filepath) {
+    if (path) {
       // strip trailing /
       url = url.replace(/\/$/, '');
+      path = decodeURIComponent(path);
```

Comment:
Need to be careful with decoding/encoding path. There are two patch kinds and two path sources, which will have different encoding:

- from url param (encoded)
- filepath from form (not encoded)
- urlpath from form (encoded)

Need to make sure that the input path is always treated correctly.

Calling decode here means that encode also needs to be called below in order to preserve the right value.